common/dllserver/dllserver.cpp

```diff
@@ -632,7 +632,7 @@ ILoadedDllEntry * DllServerBase::doLoadDll(const char * name, DllLocationType ty
     location->getDllFilename(rfile);
     StringBuffer x;
     rfile.getPath(x);
-    LOG(MCdebugInfo, unknownJob, "Loading dll (%s) from location %s", name, x.str());
+    LOG(LogCategory_ProgrammerInfo, unknownJob, "Loading dll (%s) from location %s", name, x.str());
```

Comment:
Why change? For accurate self documentation. The label prefix should properly reflect the MC prefix doesn't map
What is the guideline for "too long" and why choose brevity over readability/maintainability?
Too long to type? All IDEs I'm aware of provide intellisense type features which prevent the developer from typing out the entire label, but even so, a few extra chars is a sensible sacrifice for the sake of maintainability and search-ability.
LogCategory is too long to read? I didn't consider that a hindrance, how much weight do we want to apply to that point?
Are there any extra changes because of the new prefix? How many new changes are we willing to accept for the sake of readability?

Reply:
My concern is with the readability of the source code, particularly with the cognitive effort needed to understand it.  Try an experiment scanning some source code.

LOG(MCdebugInfo) can be read in two glances with very little delay.  It is slightly more work than DBGLOG(), but I don't lose the flow.

LOG(LogCategory_ProgrammerInfo requires 3 or 4.  With (when I read it) a significant pause on "LOG(LogCategory" before I get to the information that is interesting.  The pause is long enough to interrupt my flow in understanding.

LOG(MCprogrammerInfo) v LOG(MCdebugInfo)   is similar, not quite as intrusive.  Because Programmer is a long word I can't read that it is programmer and info in the same glance.  

Most of the time picky details of naming isn't significant enough to comment on.  But with something that is used everywhere, especially when we are planning to replace DBGLOG with LOG(...) it really needs to be right.  And by right I mean as easy to read as possible.  Consistency is important for maintainability and search-ability, spelling things out in full isn't.

Reply:
The proposed changes emphasize and improve consistency, search-ability, IntelliSense code completion, and self documentation based on explicit naming convention - all for the sake of maintainability and usability.

It seems that "spelling things out in full isn't [important]." is the actual issue and the argument I see is that reading "LogCategory_ProgrammerInfo" vs "MCdebugInfo" requires 2 to 3 extra glances...

Not sure how that was quantified, but makes one wonder if speed-scanning is being prioritized above other important aspects of maintainability such as self-documentation (which actually can speed up code understanding significantly)? 

Anyway, however you prioritize these details, if they're going to affect code reviews and productivity, I suggest they're defined/quantified and published to avoid these types of conversations.

I concede the naming convention is long, perhaps pedantic, and will condense to single-glance len.

What do you prefer:
LogCatProgmrInfo
LogCatProgInfo
LogCatDbgInfo
MCDebugInfo
LCDebugInfo
LCProgInfo




. If the extra milliseconds/seconds it might require a developer to fully grasp a variable meaning is more important than variable meaning clarity/self documentation, that's fine, but let's make it a well established best-practice for HPCC development in order to avoid these types of miscommunications.



For this particular case, I still don't see how the MC prefix helps,  the average developer, and I would prefer something like , etc.

Reply:
The priority is ease of reading.  That isn't necessarily the same as being fully descriptive.   I want to be able to look at a line of code an understand it with the least effort.  Naming variables/functions is sometimes really hard.

If something is too verbose without providing any extra information then it slows down your comprehension.  When you read text you read it in chunks, so reading

LOG(LogCategory_....

That would be the first chunk - but the LogCategory tells me nothing - since I know every log call is followed by a category.

LOG(debugInfo,

is a similar length, but tells me much more information.  It allows me to concentrate on the rest of the line - the message, which contains the details that need to be understand.

In this context I would prefer no prefix (as above) - since it is always a category, but that would pollute the namespace too much, hence a short 2 character prefix.  LC might be better than MC but they are fairly similar and any improvement doesn't justify the bulk change or need for programmers to change.

This is a slightly extreme case - in that the message categories only occur in a single context.

In this case I would vote for MCdebugInfo.  (We tend to use a lower case letter after an upper case prefix.  See examples in eclhelper.hpp, jfile.hpp etc.)